### PR TITLE
obs-ffmpeg: Implemented rtsp_transport option

### PR DESCRIPTION
**Pull request by tizbac**
This is useful to avoid artifacts when there's packet loss, many CCTV cameras try using UDP which results in artifact.

Signed-off-by: Tiziano Bacocco <tizbac2@gmail.com>

### Description
I have implented the equivalent of ffmpeg's -rtsp_transport option to allow forcing using TCP even when UDP is available.
It is useful when you have cameras that are across many hops using wireless links to avoid video corruption.
Screenshot_20210628_205840

### Motivation and Context
It is required to prevent video corruption when there's packet loss and the camera advertises UDP RTSP capability

### How Has This Been Tested?
I've tested it on linux with a dahua camera which is like 20 km from me and goes through 6 ptp links

### Types of changes

- New feature (non-breaking change which adds functionality)

### Checklist:

- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [ ] My code is not on the master branch.
- [x] The code has been tested.
- [ ] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by notr1ch** [comment]
This seems like a duplicate of #4928 and has the same problem - protocol specific options don't really belong on this dialog. Imagine if we exposed every protocol option someone might want to set - the dialog would be multiple pages.

**Comment by tizbac** [comment]
i guess showing it based on the protocol beginning like rtsp , https etc could be an option

**Comment by gxalpha** [comment]
As a side note, make sure to include the module in the commit title as well and use present tense, like "obs-ffmpeg: Implement rtsp_transport option".
Also, you don't need to add any translations except en-US.ini - they will be overwritten by Crowdin

**Comment by stopforumspam** [comment]
> This seems like a duplicate of #4928 and has the same problem - protocol specific options don't really belong on this dialog. Imagine if we exposed every protocol option someone might want to set - the dialog would be multiple pages.

how about this commit as an option, one that includes all the supported rtsp_transport options?

https://github.com/stopforumspam/obs-studio/commit/9dfb5e19a46fcf0a67939870832cb04417f15662

**Comment by notr1ch** [comment]
It has the same issue - if the user is using a HTTP stream for example, it is irrelevant to show RTSP options. A generic "Input Options" field similar to the "Input Format" field would probably make more sense for users who need access to protocol-specific options.

**Comment by stopforumspam** [comment]
well im not going to write the code to parse what could be thousands of combinations of whatever the user could enter in there, checking it so that whatever is passed wont kill ffmpeg.  they could enter a hundred  character ffmpeg filter command expecting it to work.  the ffmpeg cli parser is huge.

i'll guess we'll just never see obs support tcp for rtsp

**Comment by dodgepong** [comment]
I don't think validation is necessary. It's the user's responsibility to add parameters that actually work. If it doesn't work then the user will probably recognize it pretty fast.

**Comment by stopforumspam** [comment]
i'll see if i have time today to  add very basic (as in extremely) parsing of a free string input field, one that expects single flag (ir -e) and value keys (eg -rtsp_transport).  I'm new to C so this takes me 10 times longer than any of you

File: plugins/obs-ffmpeg/data/locale/it-IT.ini [hunk 1]
```diff
@@ -26,6 +26,7 @@ Looping="Ripeti"
 InputFormat="Formato dell'input"
 BufferingMB="Buffering della rete"
 HardwareDecode="Utilizza la decodifica hardware quando disponibile"
+ForceTCP="Forza l'utilizzo del TCP con RTSP"
```
Comment (Vainock) [inline plugins/obs-ffmpeg/data/locale/it-IT.ini:29]: Translations will be overwritten/added in the next translations push, so you should revert this change here.

**Comment by stopforumspam** [comment]
How about this?  It takes key/value pairs, space delimited, and passes them to ffmpeg.  It doesnt yet support flags (without a value) being passed as I need to look at the ffmpeg code more to see if that can be supported via the `av_dict_set` route.  the only sanity checking it does is split at the equals symbol.

I've tested this with TCP and HTTP but not UDP multicast (yet)

ffdire

**Comment by RytoEX** [comment]
I'm closing this in favor of #4967.
